src/metrics/index.js

```diff
@@ -35,13 +36,17 @@ import {
   MySegmentsCollector,
   ClientCollector
 } from './Collectors';
+import { OPTIMIZED, STANDALONE_MODE } from '../utils/constants';
 
 const log = logFactory('splitio-metrics');
+const IMPRESSIONS_REFRESH_COUNT = 300000; // 5 minutes
 
 const MetricsFactory = context => {
   let impressionsRetries = 0;
   const settings = context.get(context.constants.SETTINGS);
   const storage = context.get(context.constants.STORAGE);
+  const impressionsCounter = context.get(context.constants.IMPRESSIONS_COUNTER);
+  const shouldPushImpressionsCount = settings.sync.impressionsMode === OPTIMIZED && settings.mode === STANDALONE_MODE;
```

Comment:
Maybe we can make it "mode !== CONSUMER_MODE" to prepare for a possibly future producer behavior.